crates/dns-resolver/src/resolver.rs

```diff
@@ -517,29 +531,32 @@ impl Resolver for HickoryResolver {
                 })
             }
             Err(err) => match err.kind() {
-                ResolveErrorKind::NoRecordsFound {
-                    negative_ttl,
-                    response_code,
-                    ..
-                } => Ok(Answer {
-                    canon_name: None,
-                    records: vec![],
-                    nxdomain: *response_code == ResponseCode::NXDomain,
-                    secure: false,
-                    bogus: false,
-                    why_bogus: None,
-                    response_code: *response_code,
-                    expires: Instant::now()
-                        + Duration::from_secs(negative_ttl.unwrap_or(60) as u64),
-                }),
+                ResolveErrorKind::Proto(proto_err) => match proto_err.kind() {
```

Comment:
Might be able to profitably use `err.proto()` here?

Reply:
ah, yeah, it reduces a bit of boilerplate, thanks!